src/main/java/frc/robot/subsystems/hood/Hood.java

```diff
@@ -0,0 +1,98 @@
+package frc.robot.subsystems.hood;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.units.Angle;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+import java.util.function.Supplier;
+
+import static edu.wpi.first.units.Units.Radians;
+
+public class Hood extends SubsystemBase {
+    private static Hood INSTANCE = null;
+    private final HoodInputsAutoLogged inputs = HoodIO.inputs;
+    private final HoodIO io;
+    private final Mechanism2d mechanism2d = new Mechanism2d(
+            3,3
+    );
+    private final MechanismRoot2d root = mechanism2d.getRoot("Hood", HoodConstants.mechanism2dPose.getFirst(), HoodConstants.mechanism2dPose.getSecond());
+    private final MechanismLigament2d hood = root.append(
+        new MechanismLigament2d("Hood", HoodConstants.HoodLength, 45)
+    );
+
+    private ControlMode controlMode = ControlMode.POSITION;
+
+    /**
+     * Constructor for Hood subsystem.
+     *
+     * @param io IO of the subsystem.
+     */
+    private Hood(HoodIO io) {
+        this.io = io;
+    }
+
+    /**
+     * Gets the single instance of hoodSubsystem.
+     *
+     * @return The single instance of hoodSubsystem.
+     */
+    public static Hood getInstance() {
+        return INSTANCE;
+    }
+
+    public static void initialize(HoodIO io) {
+        INSTANCE = new Hood(io);
+    }
+
+    public MutableMeasure<Angle> getAngle() {
+        return inputs.angle;
+    }
+
+    public boolean atSetpoint() {
+        return io.atSetpoint();
+    }
+
+    public Command setAngle(Supplier<Rotation2d> angle){
+        controlMode = ControlMode.POSITION;
+        return runOnce(() -> inputs.angleSetpoint.mut_replace(angle.get().getRadians(), Radians));
+    }
+
+    public Command setPower(Supplier<Double> power){
+        controlMode = ControlMode.POWER;
+        return runOnce(() -> inputs.powerSetpoint = power.get());
+    }
+
+    public Command getResetAbsoluteEncoderCommand(){
+        setPower(() -> -0.3);
+        return runOnce(() -> io.resetAbsoluteEncoder());
```

Comment:
1. This won't work. Just calling `setPower()` will only create a command, but will not run it. You need to include the command returned by `setPower()` in a composition, followed by the reset commands.
2. Because the function your create doesn't take any parameters, you can use a method reference here. Read more [here](https://docs.wpilib.org/en/stable/docs/software/commandbased/index.html#method-references-java).
3. `0.3` is probably much too fast to zero a mechanism. Anyway, this value should be a constant.
4. The function name can be shorter, there is no need for the "get" and "command" parts.

Here is a suggestion which implements points 1 and 2:
```java
                return setPower(() -> -0.3)
                .until("some condition that checks whether the hood reached the zero position...")
                .andThen(io::resetAbsoluteEncoder);
```

Reply:
2fb3fea
8d8f25f